### PR TITLE
Adds new persistent state for devices user selection.

### PR DESCRIPTION
**Pull request by damencho**
The state about currently opened devices is filtered and not stored, where we only store when user selects a device preferences.
Also allow changing input devices for Firefox when we are not in a conference.

### CODE REVIEW

File: react/features/base/settings/reducer.js [hunk 1]
```diff
@@ -30,15 +30,31 @@ const DEFAULT_STATE = {
...
+const filterSubtree = {};
+
+// start with the default state
+Object.keys(DEFAULT_STATE).forEach(key => {
+    filterSubtree[key] = true;
+});
+
+// we want to filter these props, to not be stored as they represent
+// what is currently opened/used as devices
+filterSubtree.audioOutputDeviceId = false;
```
Comment (hristoterezov) [inline react/features/base/settings/reducer.js]: Why not put those keys in the object definition?

File: react/features/device-selection/functions.js [hunk 2]
```diff
@@ -26,20 +26,30 @@ import { toState } from '../base/redux';
...
         disableDeviceChange:
             !JitsiMeetJS.mediaDevices.isDeviceChangeAvailable(),
         hideAudioInputPreview:
             !JitsiMeetJS.isCollectingLocalStats(),
         hideAudioOutputSelect: !JitsiMeetJS.mediaDevices
                             .isDeviceChangeAvailable('output'),
-        selectedAudioInputId: settings.micDeviceId,
-        selectedAudioOutputId: getAudioOutputDeviceId(),
-        selectedVideoInputId: settings.cameraDeviceId
+        selectedAudioInputId: settings.micDeviceId || settings.userSelectedMicDeviceId,
```
Comment (hristoterezov) [inline react/features/device-selection/functions.js]: Do we really need to fallback to the user selected values? Do we have a use case when we need to do this? 

I'm wondering about the use case when you don't have a device but you have something stored from another session. Isn't this going to cause showing the user selected device from previous session as currently used?

File: react/features/base/settings/reducer.js (hunk 1 above)
Comment (damencho) [reply react/features/base/settings/reducer.js]: But we override everything from default state into the object, so there is no point of doing that.

File: react/features/device-selection/functions.js (hunk 2 above)
Comment (damencho) [reply react/features/device-selection/functions.js]: Updated so we distinguish between welcome page settings and in conference settings.

**Pull request merged**